things/postgres/things.go

```diff
@@ -89,10 +89,7 @@ func (tr thingRepository) RetrieveByID(owner, id string) (things.Thing, error) {
 
 	if err != nil {
 		empty := things.Thing{}
-		if err == sql.ErrNoRows {
-			return empty, things.ErrNotFound
-		}
-		return empty, err
+		return empty, things.ErrNotFound
```

Comment:
@vstarcevic This way, all read errors will cause the same NotFound error, which is not what we want. I'd prefer to extract error info from SQL error and check if it's invalid ID format error. Something similar to this: https://github.com/mainflux/mainflux/blob/33008140268cc3d9f5edd3394dce8cbc37f31956/things/postgres/things.go#L64

This is far from perfect because in this case, you extract semantics from the error message string, but it's better than shadowing all the errors with NotFound. Also, there is a way to check if the string is a valid UUID without trying to access a database, which is a more efficient way. Maybe adding something like `Validate()` method to IdentityProvider interface (https://github.com/mainflux/mainflux/blob/master/things/uuid/idp.go)...